cmd/format-xl.go

```diff
@@ -862,45 +721,37 @@ func saveFormatXL(storageDisks []StorageAPI, formats []*formatConfigV1) error {
 	return nil
 }
 
-// initFormatXL - save XL format configuration on all disks.
-func initFormatXL(storageDisks []StorageAPI) (err error) {
-	// Initialize jbods.
-	var jbod = make([]string, len(storageDisks))
+// Return a slice of format, to be used to format uninitialized disks.
+func newFormatXLV1(diskCount int) []*formatXLV1 {
+	var jbod = make([]string, diskCount)
 
-	// Initialize formats.
-	var formats = make([]*formatConfigV1, len(storageDisks))
+	var formats = make([]*formatXLV1, diskCount)
 
-	// Initialize `format.json`.
-	for index, disk := range storageDisks {
-		if disk == nil {
-			continue
-		}
-		// Allocate format config.
-		formats[index] = &formatConfigV1{
-			Version: formatFileV1,
-			Format:  formatBackendXL,
-			XL: &xlFormat{
-				Version: xlFormatBackendV1,
-				Disk:    mustGetUUID(),
-			},
-		}
-		jbod[index] = formats[index].XL.Disk
+	for i := 0; i < diskCount; i++ {
+		format := &formatXLV1{}
+		format.Version = formatMetaVersionV1
+		format.Format = formatBackendXL
+		format.XL.Version = formatXLVersionV1
+		format.XL.Disk = mustGetUUID()
```

Comment:
It is more readable to use the literal style of populating structs like:

```
format := &formatXLV1{
    Version: ...
    ...
}
```

The above style is repetitive on the left-hand-side. The same comment can be made in multiple places in this PR. The previous version used a more readable style.

Reply:
will fix it, will wait for other comments before pushing the fixed code.

Reply:
I have put a comment on why the current form is better as a reply to your comment for the formatFSV1 struct

Reply:
Again, I feel that anonymous struct could be avoided (because it seems to encourage a more repetitive style), but will not be holding this PR due to this.